griptape/drivers/prompt/google_prompt_driver.py

```diff
@@ -37,112 +47,198 @@ class GooglePromptDriver(BasePromptDriver):
     def try_stream(self, prompt_stack: PromptStack) -> Iterator[DeltaMessage]:
-        GenerationConfig = import_optional_dependency("google.generativeai.types").GenerationConfig
-
-        messages = self._prompt_stack_to_messages(prompt_stack)
-        response: Iterator[GenerateContentResponse] = self.model_client.generate_content(
-            messages,
-            stream=True,
-            generation_config=GenerationConfig(
-                stop_sequences=self.tokenizer.stop_sequences,
-                max_output_tokens=self.max_tokens,
-                temperature=self.temperature,
-                top_p=self.top_p,
-                top_k=self.top_k,
-            ),
+        messages = self.__to_google_messages(prompt_stack)
+        response: GenerateContentResponse = self.model_client.generate_content(
+            messages, **self._base_params(prompt_stack), stream=True
         )
 
         prompt_token_count = None
         for chunk in response:
             usage_metadata = chunk.usage_metadata
 
+            content = self.__to_prompt_stack_delta_message_content(chunk.parts[0]) if chunk.parts else None
             # Only want to output the prompt token count once since it is static each chunk
             if prompt_token_count is None:
                 prompt_token_count = usage_metadata.prompt_token_count
                 yield DeltaMessage(
-                    content=TextDeltaMessageContent(chunk.text),
-                    role=Message.ASSISTANT_ROLE,
+                    content=content,
                     usage=DeltaMessage.Usage(
                         input_tokens=usage_metadata.prompt_token_count,
                         output_tokens=usage_metadata.candidates_token_count,
                     ),
                 )
             else:
                 yield DeltaMessage(
-                    content=TextDeltaMessageContent(chunk.text),
-                    role=Message.ASSISTANT_ROLE,
-                    usage=DeltaMessage.Usage(output_tokens=usage_metadata.candidates_token_count),
+                    content=content, usage=DeltaMessage.Usage(output_tokens=usage_metadata.candidates_token_count)
                 )
 
+    def _base_params(self, prompt_stack: PromptStack) -> dict:
+        GenerationConfig = import_optional_dependency("google.generativeai.types").GenerationConfig
+        ContentDict = import_optional_dependency("google.generativeai.types").ContentDict
+        Part = import_optional_dependency("google.generativeai.protos").Part
+
+        system_messages = prompt_stack.system_messages
+        if system_messages:
+            self.model_client._system_instruction = ContentDict(
+                role="system", parts=[Part(text=system_message.to_text()) for system_message in system_messages]
+            )
+
+        return {
+            "generation_config": GenerationConfig(
+                **{
+                    # For some reason, providing stop sequences when streaming breaks native functions
```

Comment:
What's the impact on the output? Should we link or open a support case with google?

Reply:
Created and linked to [an issue](https://github.com/google-gemini/generative-ai-python/issues/446)